### PR TITLE
analysis of baseline states

### PR DESCRIPTION
**Pull request by jsadler2**
This PR is code used to look into:
* #45 
* and #48 

Specifically the code
* outputs the model hidden states to csv files 
* outputs the hidden states across 6 replicates
* makes plots to visualize the hidden state data
* pulls auxiliary data from nwis (flow, water temp, specific conductivity) for comparison with hidden states
* makes a plot that compares predicted and observed DO with water temperature

### CODE REVIEW

File: 3_visualize/src/plot_output_weights.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+sys.path.insert(0, "../../2a_model/src/models/0_baseline_LSTM/")
+
+# %%
+from model import LSTMModel
+
+# %%
+m = LSTMModel(10, 3)
+
+# %%
+m.load_weights("../../2a_model/out/models/0_baseline_LSTM/train_weights/")
```
Comment (galengorski) [inline 3_visualize/src/plot_output_weights.py:31]: I'm a little confused where these weights come from

File: 3_visualize/src/plot_output_weights.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+# %%
+data = np.load("../../2a_model/out/models/0_baseline_LSTM/prepped.npz", allow_pickle=True)
```
Comment (galengorski) [inline 3_visualize/src/plot_output_weights.py:34]: Also confused here as to what you are loading, are these input data to use for plotting

**Review by galengorski — APPROVED** [review]
This all looks good to me, I just had a few clarifying questions to make sure I understood it. I think I'm still trying to get my head around the code architecture a little bit

File: 3_visualize/src/plot_output_weights.py (hunk 1 above)
Comment (jsadler2) [reply 3_visualize/src/plot_output_weights.py:31]: These are weights produced by the snakemake workflow:
https://github.com/USGS-R/drb-do-ml/blob/cee3da5b747c035c826641e9bdbb05f91b394a81/2a_model/src/models/0_baseline_LSTM/Snakefile#L71

File: 3_visualize/src/plot_output_weights.py (hunk 2 above)
Comment (jsadler2) [reply 3_visualize/src/plot_output_weights.py:34]: Yes these are input data. They are used to just run the model once so I can get the weight values. ... now that I think of it, I'm not 100% sure this step is necessary.

**Pull request merged**